### PR TITLE
ssl: inject options into transport setup

### PR DESCRIPTION
**Pull request by robertkowalski**
This makes the setup configuration available to the transport
constructor

### CODE REVIEW

File: lib/Peer.js [hunk 1]
```diff
@@ -20,7 +20,8 @@ class Peer extends Events {
   }
 
   transport (dest) {
-    return this.newTransport({ dest: dest })
+    const opts = Object.assign({}, this.conf, { dest: dest })
```
Comment (prdn) [inline lib/Peer.js]: @robertkowalski can we use lodash `extend` or `defaults` (it's just to keep things uniform)

File: lib/TransportRPC.js [hunk 2]
```diff
@@ -15,6 +15,13 @@ class TransportRPC extends TransportSocket {
   format (data) {
     return JSON.stringify(data)
   }
+
+  getSocket (port, conf) {
+    // https://github.com/uWebSockets/bindings/issues/23
+    throw new Error(
+      'implementation error: getSocket method needs to get extended'
+    )
+  }
```
Comment (robertkowalski) [inline lib/TransportRPC.js:24]: @prdn i added the `getSocket` method to `TransportRPC.js` to have it available in the inheriting `TransportRPCServer.js` classes

Comment (robertkowalski) [reply lib/TransportRPC.js:24]: ah wait, thats odd. i'll change that

Comment (robertkowalski) [reply lib/TransportRPC.js:24]: hmm, not super happy with `getSocket` but it also doesn't make sense to me to move it into the Peer*.js classes, as they are basically factories for the different transports...

🤔

Comment (robertkowalski) [reply lib/TransportRPC.js:24]: ah, i just get the socket and inect it as part of the factory method... on it...

File: lib/TransportRPCClient.js [hunk 3]
```diff
@@ -5,9 +5,11 @@ const uuid = require('uuid')
 const TransportRPC = require('./TransportRPC')
 
 class TransportRPCClient extends TransportRPC {
-  constructor (client, conf) {
+  constructor (client, conf, Socket) {
```
Comment (prdn) [inline lib/TransportRPCClient.js]: @robertkowalski may I ask you why you pass the class here and not directly the instance?
Having it as a parameter (uppercase) in the constructor is not striking me to be...right

File: lib/TransportRPCServer.js [hunk 4]
```diff
@@ -3,6 +3,12 @@
 const TransportRPC = require('./TransportRPC')
 
 class TransportRPCServer extends TransportRPC {
+  constructor (client, conf, Socket) {
+    super(client, conf)
+
+    this.Socket = Socket
```
Comment (prdn) [inline lib/TransportRPCServer.js]: @robertkowalski isn't it better to move the `getSocket` inside the transport class?

File: lib/TransportRPCClient.js (hunk 3 above)
Comment (robertkowalski) [reply lib/TransportRPCClient.js]: To create an instance we need often options specific to the implementation (destination for example).

Also right now we know the port very late in the bootstrap, as it is passed to the `.listen` method of the rpc-server:

https://github.com/robertkowalski/grenache-nodejs-ws/blob/33e9d82fd6c107073617fab767f3a9e633a44cca/lib/TransportRPCServer.js#L6-L9

File: lib/TransportRPCServer.js (hunk 4 above)
Comment (robertkowalski) [reply lib/TransportRPCServer.js]: haha i had that before, removed it this afternoon, as you pointed me to `PeerRPC.js` this morning so i thought i must have missed something:

screen shot 2017-05-26 at 01 39 06

after thinking about it again you are right, and it should also solve the problem described in the other comment

Comment (robertkowalski) [reply lib/TransportRPCServer.js]: @prdn i still have the other commit in my reflog. 

i can easily revert to the transport-getSocket version, what do you think?

Comment (prdn) [reply lib/TransportRPCServer.js]: yeah I prefer to have the getSocket in the transport :)

File: lib/TransportRPCClient.js (hunk 3 above)
Comment (prdn) [reply lib/TransportRPCClient.js]: this should be solved as soon as we move the getSocket in the transport

**Pull request merged**